File: tests/benchmarks/tight_loop/tight_loop.py
import time

def tight_loop():
    i = 0
    sum = 0
    while i < 10000000:
        sum = sum + i
        i = i + 1
    return sum

start = time.time()
print(tight_loop())
end = time.time()
print(f"Python Time: {end - start:.4f}s")

File: tests/benchmarks/tight_loop/tight_loop.js
function tightLoop() {
    let i = 0;
    let sum = 0;
    while (i < 10000000) {
        sum = sum + i;
        i = i + 1;
    }
    return sum;
}

const start = performance.now();
const result = tightLoop();
const end = performance.now();
console.log(result);
console.log(`JS Time: ${((end - start) / 1000).toFixed(4)}s`);
